riscv/src/compiler.rs

```diff
@@ -1259,7 +1387,7 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
             };
             match (replacement, instr) {
                 (Some(replacement), "call") => vec![replacement],
-                (Some(replacement), "tail") => vec![replacement, "ret".to_string()],
+                (Some(replacement), "tail") => vec![replacement, "ret;".to_string()],
```

Comment:
Can you extract that? Then we can merge it right away.

Reply:
Sure, see #537